src/tools/schema2kotlin.ts

```diff
@@ -72,33 +73,36 @@ class KotlinGenerator implements ClassGenerator {
   fieldSets: string[] = [];
   encode: string[] = [];
   decode: string[] = [];
+  fieldsReset: string[] = [];
   getUnsetFields: string[] = [];
 
   constructor(readonly node: SchemaNode) {}
 
   addField(field: string, typeChar: string) {
     const {type, decodeFn, defaultVal} = typeMap[typeChar];
     const fixed = field + (keywords.includes(field) ? '_' : '');
+    const set = `_${fixed}Set`;
 
     this.fields.push(`${fixed}: ${type}`);
     this.fieldVals.push(
-      `var _${fixed}Set = false\n` +
+      `var ${set} = false\n` +
       `    var ${fixed} = ${defaultVal}\n` +
       `        get() = field\n` +
       `        set(value) {\n` +
       `            field = value\n` +
-      `            _${fixed}Set = true\n` +
+      `            ${set} = true\n` +
       `        }`
     );
     this.setFields.push(`this.${fixed} = ${fixed}`);
-    this.fieldSets.push(`_${fixed}Set`);
-    this.getUnsetFields.push(
-      `if(!_${fixed}Set) {\n` +
-      `             rtn.add("${fixed}")\n` +
-      `         }`
+    this.fieldSets.push(`${set}`);
+    this.fieldsReset.push(
+      `${fixed} = ${defaultVal}`,
+      `${set} = false`
     );
+    this.getUnsetFields.push(`if (!${set}) rtn.add("${fixed}")`);
 
     this.decode.push(`"${field}" -> {`,
+                     `    if (${set}) log("WARNING: there are multiple fields ${field}")`,
```

Comment:
I don't think we want this. It isn't really a problem for the entity encoding to repeat a field value.

Reply:
Updated